incorrect treatment of multiple informational HTTP responses

NOTE: this is not the same as <https://github.com/square/okhttp/issues/7440>.

From <https://github.com/greenbytes/java-http-1xx-tests>:

~~~
mvn clean install -Dtest=OkHttpClientTests
~~~

Responses that contain *multiple* 1xx messages are treated incorrectly:

~~~
[ERROR] testOkHttpClient104_100(org.greenbytes.http.test1xx.OkHttpClientTests)  Time elapsed: 0.459 s  <<< ERROR!
java.lang.IllegalStateException: state: 3
        at org.greenbytes.http.test1xx.OkHttpClientTests.testOkHttpClient(OkHttpClientTests.java:20)
        at org.greenbytes.http.test1xx.OkHttpClientTests.testOkHttpClient104_100(OkHttpClientTests.java:63)

[ERROR] testOkHttpClient104_2(org.greenbytes.http.test1xx.OkHttpClientTests)  Time elapsed: 0.006 s  <<< ERROR!
java.lang.IllegalStateException: state: 3
        at org.greenbytes.http.test1xx.OkHttpClientTests.testOkHttpClient(OkHttpClientTests.java:20)
        at org.greenbytes.http.test1xx.OkHttpClientTests.testOkHttpClient104_2(OkHttpClientTests.java:58)
~~~

--- a/okhttp/src/main/kotlin/okhttp3/internal/http/CallServerInterceptor.kt
+++ b/okhttp/src/main/kotlin/okhttp3/internal/http/CallServerInterceptor.kt
@@ -105,7 +105,7 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
           .build()
       var code = response.code
 
-      if (shouldIgnoreAndWaitForRealResponse(code, exchange)) {
+      while (shouldIgnoreAndWaitForRealResponse(code, exchange)) {
         responseBuilder = exchange.readResponseHeaders(expectContinue = false)!!
         if (invokeStartEvent) {
           exchange.responseHeadersStart()

--- a/okhttp/src/test/java/okhttp3/CallTest.kt
+++ b/okhttp/src/test/java/okhttp3/CallTest.kt
@@ -3235,6 +3235,33 @@ open class CallTest {
   }
 
   @Test
+  fun serverRespondsWithProcessingMultiple() {
+    server.enqueue(
+      MockResponse.Builder()
+        .apply {
+          repeat(10) {
+            addInformationalResponse(
+              MockResponse(
+                code = HTTP_PROCESSING,
+              ),
+            )
+          }
+        }
+        .build(),
+    )
+    val request =
+      Request(
+        url = server.url("/"),
+        body = "abc".toRequestBody("text/plain".toMediaType()),
+      )
+    executeSynchronously(request)
+      .assertCode(200)
+      .assertSuccessful()
+    val recordedRequest = server.takeRequest()
+    assertThat(recordedRequest.body.readUtf8()).isEqualTo("abc")
+  }
+
+  @Test
   fun serverRespondsWithUnsolicited100Continue_HTTP2() {
     enableProtocol(Protocol.HTTP_2)
     serverRespondsWithUnsolicited100Continue()

